src/content/learn/fetching-data.md

```diff
@@ -0,0 +1,206 @@
+
+```js
+async function LATemperature() {
+  const { temperature } = await fetch(
+    `https://api.open-meteo.com/v1/forecast?latitude=52.52&longitude=13.41&hourly=temperature_2m`,
+  );
+  return <p>The current temperature in Los Angeles is: {temp}.</p>;
+}
+```
+
+An API service will usually provide documentation on how to make a request to their server and how the data is returned to you.
+
+## Waiting for asynchronous data during render {/*waiting-for-data-during-render*/}
+
+You'll notice that once we started to read data during render, we began to use the keywords `async` and `await` in our component. Let's dive deeper into what that means.
+
+```js
+import { readFile } from "node:fs/promises";
+
+async function LATemperature() {
+  const tempPromise = readFile("/path/to/temperature");
+  const temp = await tempPromise;
+  return <p>The current temperature in LA is: {temp}</p>;
+}
+```
+
+In our local file example, we imported the function `readFile` from a file streaming package. `readFile` is an asynchronous function that takes in a file path and returns the file contents. An asynchronous function means that JavaScript does not pause execution to wait for the function to finish. Instead, it returns a Promise, a container for its future return value.
+
+An analogy to calling an asynchronous function is like taking your clothes to the dry cleaners. You've given the work to clean your clothes to someone else, leaving you free to do other things. The dry cleaners give you a ticket, a promise, that they will complete this work and notify you when they're done. When the work is done successfully, the Promise is considered resolved.
+
+<Illustration>
+Illustration of the parallel work while getting dry cleaning
+</Illustration>
+
+Now, imagine you are trying to render the component `LATemperature`. The first thing you do is kick off the asynchronous work of reading the temperature file and you've received `tempPromise` as a Promise. Now, you're free to do the next set of work. But wait, the next instruction requires you to know the temperature, yet all we have is a Promise. The Promise may or may not be ready with the actual temperature value.
+
+Here is where `await` comes in. `await` is a keyword that is used before a Promise and tells JavaScript to pause running the current function and wait for the Promise to resolve.Then, when it is resolved, await will unwrap the Promise container to return the value. We use await on tempPromise to pause our component rendering until the file is read and receive the value returned.
+
+In our analogy, this would be similar to waiting at the dry cleaners for them to finish because the next chore on your todo list requires those clothes. Thus, preventing you from doing anything else in the meantime.
```

Comment:
The analogy seems imperfect: This makes it sound like using `await` is equivalent to using a synchronous API that pauses the main thread. 

Maybe you need to introduce another task that depends on the dry cleaning, like packing your suitcase for a trip. You can do other stuff in the meantime, but to finish packing your suitcase you need to get the dry cleaning back.